core/src/main/java/io/substrait/expression/WindowBound.java

```diff
@@ -5,58 +5,62 @@
 @Value.Enclosing
 public interface WindowBound {
 
-  public BoundedKind boundedKind();
+  interface WindowBoundVisitor<R, E extends Throwable> {
+    R visit(Preceding preceding);
 
-  enum BoundedKind {
-    UNBOUNDED,
-    BOUNDED,
-    CURRENT_ROW
-  }
+    R visit(Following following);
+
+    R visit(CurrentRow currentRow);
 
-  enum Direction {
-    PRECEDING,
-    FOLLOWING
+    R visit(Unbounded unbounded);
   }
```

Comment:
Direction doesn't have an equivalent in the Substrait spec itself. Calcite has a concept of `UNBOUNDED_PRECEDING` and `UNBOUNDED_FOLLOWING`, but those can be reduced to singular `Unbounded` WindowBound when converting from Calcite to Pojo.